packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -60,13 +73,128 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody> {
       axis: Axis.horizontal,
       initialFirstFraction: 0.25,
       // TODO(https://github.com/flutter/devtools/issues/1648): Debug panes.
-      firstChild: const Text('Debugger details'),
+      firstChild: ScriptPicker(
+        scripts: scriptList,
+        onSelected: onScriptSelected,
+        selected: loadingScript,
+      ),
       // TODO(https://github.com/flutter/devtools/issues/1648): Debug controls.
-      secondChild: CodeView(
-        script: script,
+      secondChild: loadingScript != null && script == null
+          ? const Center(child: CircularProgressIndicator())
+          : CodeView(
+              script: script,
+            ),
+    );
+  }
+}
+
+/// Picker that takes a [ScriptList] and allows selection of one of the scripts inside.
+class ScriptPicker extends StatefulWidget {
+  const ScriptPicker({Key key, this.scripts, this.onSelected, this.selected})
+      : super(key: key);
+
+  final ScriptList scripts;
+  final void Function(ScriptRef scriptRef) onSelected;
+  final ScriptRef selected;
+
+  @override
+  ScriptPickerState createState() => ScriptPickerState();
+}
+
+class ScriptPickerState extends State<ScriptPicker> {
+  List<ScriptRef> _filtered;
+  TextEditingController filterController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    if (_isNotLoaded) initFilter();
+  }
+
+  @override
+  void didUpdateWidget(ScriptPicker oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_isNotLoaded) {
+      initFilter();
+    } else if (oldWidget.scripts != widget.scripts) {
+      updateFilter(filterController.text);
+    }
+  }
+
+  void initFilter() {
+    // Make an educated guess as to the main package to slim down the initial list of scripts we show.
+    if (widget.scripts?.scripts != null) {
+      final mainFile = widget.scripts.scripts
+          .firstWhere((ref) => ref.uri.contains('main.dart'));
+      filterController.text = mainFile.uri.split('/').first;
+      updateFilter(filterController.text);
+    }
+  }
+
+  void updateFilter(String value) {
+    setState(() {
+      if (widget.scripts?.scripts == null) {
+        _filtered = null;
+      } else {
+        _filtered = widget.scripts.scripts
+            .where(
+              (ref) => ref.uri.contains(value.toLowerCase()),
```

Comment:
nit: consider removing this trailing comma.

Reply:
this logic should be extended to match what was in html_scripts_view.
specifically, use getShortScriptName or  equivalent before calling contains.

Reply:
I've left a TODO to do this.  Currently debugger_state is not importable in flutter code.

Reply:
sg